### ISSUE TITLE
Exception with github login

### ISSUE THREAD

**Issue by mms-segu** [sandstorm-io/sandstorm#1022]
Hi,

I installed the latest sandstorm version on a vm with the default settings (except https).
I enabled access via Github.
When I click the invite-link, I get the following message: 
```
You've been invited to join this Sandstorm server! First, you have to sign in. Click "Sign in" in the upper-right.```
When trying to login with Github, I get redirected to github, authorize sandstorm and get redirected back to my sandstorm url.
```
There, nothing happens, except for the following error in the log:
```
{"line":"398","file":"oauth_server.js","message":"Error in OAuth Server: redirectUrl (http://URL:6080/) is not on the same host as the app (http://URL:6080/)","time":{"$date":1444814611722},"level":"warn"}
Exception while invoking method 'login' Error: Failed to complete OAuth handshake with Github. connect ECONNREFUSED
    at getAccessToken (packages/github/github_server.js:46:1)
    at Object.userAgent [as handleOauthRequest] (packages/github/github_server.js:5:1)
    at OAuth._requestHandlers.(anonymous function) (packages/oauth2/oauth2_server.js:8:1)
    at middleware (packages/oauth/oauth_server.js:173:1)
    at packages/oauth/oauth_server.js:146:1
```

URL is a FQDN.

What is the problem? The Connection Refused, or the same host problem?

```
SERVER_USER=sandstorm
PORT=6080
MONGO_PORT=6081
BIND_IP=0.0.0.0
BASE_URL=http://URL:6080
WILDCARD_HOST=*.URL:6080
UPDATE_CHANNEL=dev
ALLOW_DEV_ACCOUNTS=false
```

**Comment by zarvox** [comment]
After you, the user, complete the OAuth login flow, GitHub redirects you back to sandstorm with a token in the URL.  Sandstorm takes that token and asks GitHub "a user gave me this token, are they authenticated, and if so, as whom?" over an outgoing HTTP request.

It looks like that request from your Sandstorm server to GitHub was either blocked by an outbound firewall, or GitHub's API was temporarily down.  Do you have a firewall that blocks outbound connections?

**Comment by paulproteus** [comment]
Hi @mms-segu ,

Did this issue persist? If so, then perhaps we should adjust something in Sandstorm to show a more useful error message in the case that Github.com can't be reached from the Sandstorm side.

**Comment by mms-segu** [comment]
@paulproteus, sorry for not answering!
It was indeed a blocking firewall! No need to do something here.

**Comment by paulproteus** [comment]
Thanks @mms-segu for filing this. I filed a follow-up ticket that would improve Sandstorm in a way that would hopefully have avoided the need for you to file this ticket.

https://github.com/sandstorm-io/sandstorm/issues/1429
